lacework/resource_lacework_integration_ghcr.go

```diff
@@ -0,0 +1,272 @@
+			},
+			"created_or_updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"org_level": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceLaceworkIntegrationGhcrCreate(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	data := api.NewContainerRegistry(d.Get("name").(string),
+		api.GhcrContainerRegistry,
+		api.GhcrData{
+			LimitByTag:            castAttributeToStringSlice(d, "limit_by_tags"),
+			LimitByLabel:          castAttributeToArrayOfKeyValueMap(d, "limit_by_label"),
+			LimitByRep:            castAttributeToStringSlice(d, "limit_by_repositories"),
+			LimitNumImg:           d.Get("limit_num_imgs").(int),
+			RegistryDomain:        "ghcr",
+			RegistryNotifications: d.Get("registry_notifications").(bool),
+			Credentials: api.GhcrCredentials{
+				Username: d.Get("credentials.0.username").(string),
+				Password: d.Get("credentials.0.password").(string),
+				Ssl:      d.Get("credentials.0.ssl").(bool),
+			},
+		},
+	)
+
+	if !d.Get("enabled").(bool) {
+		data.Enabled = 0
+	}
+
+	log.Printf("[INFO] Creating ContVulnCfg integration for %s registry type with data:\n%+v\n",
+		api.GhcrContainerRegistry.String(), data)
+	response, err := lacework.V2.ContainerRegistries.Create(data)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.IntgGuid)
+	d.Set("name", response.Data.Name)
+	d.Set("intg_guid", response.Data.IntgGuid)
+	d.Set("enabled", response.Data.Enabled == 1)
+	d.Set("created_or_updated_time", response.Data.CreatedOrUpdatedTime)
+	d.Set("created_or_updated_by", response.Data.CreatedOrUpdatedBy)
+	d.Set("type_name", response.Data.Type)
+	d.Set("org_level", response.Data.IsOrg == 1)
+
+	log.Printf("[INFO] Created ContVulnCfg integration for %s registry type with guid %s\n",
+		api.GhcrContainerRegistry.String(), response.Data.IntgGuid)
+	return nil
+}
+
+func resourceLaceworkIntegrationGhcrRead(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	log.Printf("[INFO] Reading ContVulnCfg integration for %s registry type with guid %s\n",
+		api.GhcrContainerRegistry.String(), d.Id())
+	response, err := lacework.V2.ContainerRegistries.GetGhcr(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", response.Data.Name)
+	d.Set("intg_guid", response.Data.IntgGuid)
+	d.Set("enabled", response.Data.Enabled == 1)
+	d.Set("created_or_updated_time", response.Data.CreatedOrUpdatedTime)
+	d.Set("created_or_updated_by", response.Data.CreatedOrUpdatedBy)
+	d.Set("type_name", response.Data.Type)
+	d.Set("org_level", response.Data.IsOrg == 1)
+
+	creds := struct {
+		username string
+		password string
+		ssl      bool
+	}{username: response.Data.Data.Credentials.Username,
+		password: response.Data.Data.Credentials.Password,
+		ssl:      response.Data.Data.Credentials.Ssl}
+	d.Set("credentials", creds)
```

Comment:
does this actually work @dmurray-lacework ? did you test this? I normally use a `map[string]string` 🤔

Reply:
As far as I can tell yes. It is tested. Happy to change it for consistency elsewhere. Ssl is a bool, we couldn't use map[string]string here.

Reply:
I see, yeah that makes sense. Well now you don't have a `credentials` block, so it doesn't make sense to store it in the state, you can store everything at the top-level schema. You can do:
```go
d.Set("username", response.Data.Data.Credentials.Username
d.Set("password", response.Data.Data.Credentials.Password)
d.Set("ssl", response.Data.Data.Credentials.Ssl)

```